diff --git a/win/makefile.vc b/win/makefile.vc
index 6c9dd624d7..d29185feed 100644
--- a/win/makefile.vc
+++ b/win/makefile.vc
@@ -420,6 +420,7 @@ TCLSTUBOBJS = \
 ### the left side of implicit rules.
 TOMMATHDIR	= $(ROOT)\libtommath
 PKGSDIR		= $(ROOT)\pkgs
+WINDIR		= $(ROOT)\win
 
 # Additional include and C macro definitions for the implicit rules
 # defined in rules.vc
@@ -438,6 +439,7 @@ TESTFLAGS = $(TESTFLAGS) -file $(TESTPAT)
 !endif
 
 
+
 #---------------------------------------------------------------------
 # Project specific targets
 # There are 4 primary build configurations to consider from the combination
@@ -466,18 +468,24 @@ TESTFLAGS = $(TESTFLAGS) -file $(TESTPAT)
 # release   - Everything that builds as part of a release
 #---------------------------------------------------------------------
 
-release:    setup $(TCLSH) $(TCLSTUBLIB) dlls pkgs tk-build
-all:	    setup $(TCLSH) $(TCLSTUBLIB) dlls $(CAT32) pkgs
+release:    setup $(TCLSH) $(TCLSTUBLIB) dlls setpath pkgs tk-build
+all:	    setup $(TCLSH) $(TCLSTUBLIB) dlls $(CAT32) setpath pkgs
 core:	    setup $(TCLLIB) $(TCLSTUBLIB)
 shell:	    setup $(TCLSH)
 dlls:	    setup $(TCLREGLIB) $(TCLDDELIB) $(OUT_DIR)\zlib1.dll
 tcltest:    setup $(TCLTEST) dlls $(CAT32)
-install:    install-binaries install-libraries install-docs install-pkgs tk-build tk-install
+install:    setpath install-binaries install-libraries install-docs install-pkgs tk-build tk-install
 !if $(SYMBOLS)
 install:    install-pdbs
 !endif
 setup:      default-setup
 
+# Add TCL win directory to PATH for current session
+setpath:
+	@echo Adding $(WINDIR) to PATH
+	@set PATH=$(WINDIR);$(PATH)
+	@echo PATH updated for current session
+
 test: test-core test-pkgs
 test-core: setup $(TCLTEST) dlls $(CAT32)
 	set TCL_LIBRARY=$(ROOT:\=/)/library
-- 
2.47.1.windows.2

